### ISSUE TITLE
`show.velocity.on.embedding.cor` calculations expensive - separate function?

### ISSUE THREAD

**Issue by robertamezquita** [velocyto-team/velocyto.R#15]
When testing different visual parameters, it is pretty expensive to continually recalculate the delta projects, sqrt knn, transition probs, and such. Is there a way to save the results of these calculations or create a separate function for their calculation and reuse them for iterating on a plot more rapidly? 

Also, for this function's returns, it might be nice to have the resulting ggplot/base data be returned so that the results can be ported over to the user's desired graphical interface (for example, if I want to use transpose the arrows onto a different ggplot of my making).

**Comment by andrewhill157** [comment]
+1

**Comment by pkharchenko** [comment]
we'll try to move this to ggplot in the next release, meanwhile, you can skip the most expensive recalculation by passing back cc parameter:

x <- show.velocity.on.embedding.cor( … );
show.velocity.on.embedding.cor( … , cc=x$cc)

-peter.

**Comment by lambdamoses** [comment]
Yes, using `cc = x$cc` helps, but it still calculates knn, transition probs, arrows, and etc. every single time, which still takes a while, though not nearly computationally expensive as the delta projection step. Sometimes I simply want to change the size of the arrows and colors of the cells, that's it, and recomputing knn, transition probs, etc. are not relevant. Can you store all the calculated parts somewhere, or use memoization, so nothing is recomputed when changing plot appearances that are not relevant to calculations?

**Comment by lambdamoses** [comment]
Another suggestion for `show.velocity.on.embedding.cor`: it won't take significant change and should be pretty easy to implement. Can you give us options to customize point characters and arrow colors?

**Comment by EugeneEA** [comment]
Hi, thanks for your efforts in maintaining and developing velocito. I'd like to ask - was this issue resolved? I also agree that it is extrimely inconvinient to re-calculate everything to change plotting paramters. As far as I understand I can now pass 
`return.details = TRUE`
to "show.velocity.on.embedding.cor". But are there any code examples for using this output outside "pagoda" just with bare gglot2? (or internal plotting functions) 

Best, Eugene

**Comment by yamihn** [comment]
@robertamezquita hello, once I saved the show.velocity.on.embedding.cor into a variable plot= show.velocity.on.embedding.cor..
how can I visualize the plot? Thank you

**Comment by andrei-stoica26** [comment]
@yamihn 
You can convert the `show.velocity.on.embedding.cor` plot to a ggplot object by using `gridGraphics` and `ggplotify`.

First, load `gridGraphics` and create a function as follows:

```
GrabGrob <- function(){
  grid.echo()
  grid.grab()
}
```
Then just type `p <- GrabGrob()` right after your `show.velocity.on.embedding.cor`. This simply captures the graphics currently shown in RStudio, hence it will not need `return.details` to be set to `TRUE` in `show.velocity.on.embedding.cor`.

Now you can use `ggplotify::asggplot` to convert the plot to a ggplot object: `p <- as.ggplot(p)`.

**Comment by yamihn** [comment]
Thank you very much, you are very kind!
